### PR TITLE
Replace common.sh with Rust code

### PR DESCRIPTION
**Pull request by ice1000**
Signed-off-by: ice1000 <ice1000kotlin@foxmail.com>

So we have fewer files in the root directory in this repo and no longer depend on `sed`.

### CODE REVIEW

**Comment by ice1000** [comment]
Ooops, I didn't know `generate_go.sh` depends on this too...

**Comment by ice1000** [comment]
I'm fine with closing this but I'd like to listen to others' opinions.

**Comment by nrc** [comment]
seems like a win to me?

**Comment by Hoverbear** [comment]
This should fix #355

File: build.rs [hunk 1]
```diff
@@ -50,21 +51,26 @@ fn main() {
...
+    let caps = ver_re
+        .captures(str::from_utf8(&ver.stdout).unwrap())
+        .unwrap();
+    let major = caps.get(1).unwrap().as_str().parse::<i16>().unwrap();
+    let minor = caps.get(2).unwrap().as_str().parse::<i16>().unwrap();
+    if major == 3 && minor < 1 || major < 3 {
         panic!(
-            "Invalid version of protoc (required 3.1.x), or protoc not installed\n\nstdout:\n\n{}",
-            String::from_utf8_lossy(&output.stdout)
+            "Invalid version of protoc (required 3.1.x, get {}.{}.x).",
```
Comment (Hoverbear) [inline build.rs]:
```suggestion
            "Invalid version of protoc (required 3.1.x, got {}.{}.x).",
```

**Comment by ice1000** [comment]
PTAL @nrc

**Pull request merged**